op-supervisor/supervisor/backend/l1access/l1_accessor.go

```diff
@@ -159,7 +159,26 @@ func (p *L1Accessor) onFinalized(ctx context.Context, ref eth.L1BlockRef) {
 }
 
 func (p *L1Accessor) onLatest(ctx context.Context, ref eth.L1BlockRef) {
-	p.tipHeight = ref.Number
+	// Stop if the block is the same or older than the tip
+	if ref.Number == p.tip.Number && ref.Hash == p.tip.Hash {
+		p.log.Info("Latest L1 block signal is the same as the tip", "ref", ref)
+		return
+	}
+	if ref.Number < p.tip.Number {
+		p.log.Warn("L1 block is older than the tip", "ref", ref)
+		return
+	}
+
+	// If the incoming block is not the child of the current tip, signal a potential reorg
+	if ref.ParentHash != p.tip.Hash {
+		p.emitter.Emit(superevents.RewindL1Event{
+			IncomingBlock: ref.ID(),
+		})
+		p.log.Info("Reorg detected", "ref", ref)
+	}
```

Comment:
This will also trigger if somehow `onLatest` missed an L1 while following the `tip`. I'm assuming the "potential reorg" gets determined unnecessary if all the data in the db is correct, so I guess this makes sense.

And then you always set the tip, even after informing of the potential reorg, which I think also makes sense.